intel_extension_for_transformers/llm/runtime/graph/core/layers/mha_dense.cpp

```diff
@@ -2006,14 +2018,39 @@ void jblas_reordered_attn_fp32_update_v(const jblas_fusion_attn_fp32_update_kv_a
             const auto i_blk = i_dst - ii;
             const auto jj = j % 48;
             const auto j_blk = j - jj;
-            dst[i_blk * 48 + ii + j_blk * pad_seq_max + jj * 2] =
-                i < p.seq_size ? static_cast<bf16>(src[i * p.step_seq + j]) : bf16(0);
+            if constexpr (zero_padding) {
+              dst[i_blk * 48 + ii + j_blk * pad_seq_max + jj * 2] =
+                  i < p.seq_size ? static_cast<bf16>(src[i * p.step_seq + j]) : bf16(0);
+            } else {
+              if (i < p.seq_size)
+                dst[i_blk * 48 + ii + j_blk * pad_seq_max + jj * 2] = static_cast<bf16>(src[i * p.step_seq + j]);
+            }
           }
         }
       }
     }
   }
 }
+void jblas_reordered_attn_fp32_update_v(const jblas_fusion_attn_fp32_update_kv_args_t* params) {
+  return params->no_zeroing ? jblas_reordered_attn_fp32_update_v_<false>(params)
+                            : jblas_reordered_attn_fp32_update_v_<true>(params);
+}
+
+void jblas_reordered_attn_fp32_shift_rope_k(char* cache, const ne_fp16_t* cossin, int batch_size, int heads_kv,
+                                            int head_size, int seq_max, int seq_keep) {
+  const auto pad_headsize = padto(head_size, 32);
+  const auto pad_seq_max = padto(seq_max, 48);
+  const auto cache_step_head_num = pad_headsize * pad_seq_max;
+  const auto cache_step_bs = heads_kv * cache_step_head_num;
+
+#pragma omp parallel for collapse(2)
+  for (int ibs = 0; ibs < batch_size; ++ibs)
+    for (int ihn = 0; ihn < heads_kv; ++ihn) {
+      const auto src = reinterpret_cast<bf16*>(cache) + ibs * cache_step_bs + ihn * cache_step_head_num;
+      jblas::kernel::jit::CScaleInterleavedBF16FP16::forward<48>(src, reinterpret_cast<const fp16*>(cossin), head_size,
```

Comment:
calling jit code directly violates our common practice, don't know whether this is expected after new API @luoyu-intel

Reply:
the whole mha_dense is enabled only under amx_bf16 ISA. so it has no ISA combination.